internal/api/middleware.go

```diff
@@ -0,0 +1,56 @@
+package api
+
+import (
+	"log/slog"
+	"net/http"
+	"os"
+
+	"github.com/felixge/httpsnoop"
+)
+
+var Log *slog.Logger
+var lvl *slog.LevelVar
+
+var reqID = 0
+
+func init() {
+	lvl = new(slog.LevelVar)
+	lvl.Set(slog.LevelDebug)
+	Log = slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{Level: lvl}))
+}
+
+func loggingMiddleware(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		reqID++
+		Log.Info(
+			"REQ",
+			slog.Int("id", reqID),
```

Comment:
Для уточнения какой это все-таки id стоит дать более информативное имя. 

```suggestion
			slog.Int("request_id", reqID),
```